### PR TITLE
Issue #299: unclear error message in case of multiline string argument

### LINKED ISSUE #299: Multiline strings in mutation error
Hello,
I have an issue whereby sending a multiline string in a mutation will return a difficult to understand error message.
I have reproduced this issue on the starwars example in this repo.

**Graphql mutation(not using block strings) that will reproduce the error:**
```
mutation {
  createReview(episode:NEWHOPE, review: {stars: 0, commentary: "1
    2"}) {
    
  }
}
```
**Expected:** Error message indicating that a multiline/un-terminated string was found in the query.
**Actual:** 
```
{
  "errors": [
    {
      "message": "syntax error: unexpected \"2\", expecting Ident",
      "locations": [
        {
          "line": 32,
          "column": 5
        }
      ]
    }
  ]
}
```
Log output: `<input>:31:64: literal not terminated `

### PR DESCRIPTION
**Pull request by suntoucha**
Fix #299 

Package text/scanner, which is used to parse queries, already detects multiline strings as an error. But it only puts log message to console, as it was pointed in the issue description. Scanner can also call a function each time it encounters error 

So the idea of the fix is:
- provide Scanner a function to catch errors
- generate high-level errors in that function
- couple of tests

### CODE REVIEW

**Comment by suntoucha** [comment]
Excuse me, can anyone give a hint what could be the difference between local test run and semaphoreci?

Local run passes ok:

> --- PASS: TestMultilineString (0.00s)
>     --- PASS: TestMultilineString/Oneline_strings_are_okay (0.00s)
>     --- PASS: TestMultilineString/Multiline_strings_are_not_allowed (0.00s)

Semaphoreci failes (part of the returned error string after "See near..." is missing):

> --- FAIL: TestMultilineString (0.00s)
>     --- PASS: TestMultilineString/Oneline_strings_are_okay (0.00s)
>     --- FAIL: TestMultilineString/Multiline_strings_are_not_allowed (0.00s)
>     	lexer_test.go:126: Test 'Multiline strings are not allowed' failed with wrong error: 'graphql: syntax error: literal not terminated. See near  (line 1, column 1)'. Error should be: 'graphql: syntax error: literal not terminated. See near "Hello (line 1, column 1)'

Thank you!

**Comment by suntoucha** [comment]
Tests are OK now

The only difference I've noticed is that SemaphoreCI runs on go 1.11 by default

File: internal/common/lexer.go [hunk 1]
```diff
@@ -219,3 +223,7 @@ func (l *Lexer) consumeComment() {
 		l.comment.WriteRune(next)
 	}
 }
+
+func (l *Lexer) CatchScannerError(s *scanner.Scanner, msg string) {
+	l.SyntaxError(msg)
+}
```
Comment (pavelnikolov) [inline internal/common/lexer.go:229]: Add empty line at the end.

File: internal/common/lexer_test.go [hunk 2]
```diff
@@ -93,3 +93,43 @@ func TestConsume(t *testing.T) {
...
+					t.Fatalf("Test '%s' failed with wrong error: '%s'. Error should be: '%s'", test.description, err.Error(), test.expected)
+				}
+			}
+
+			if !test.failureExpected && err != nil {
+				t.Fatalf("Test '%s' failed with error: '%s'", test.description, err.Error())
+			}
+		})
+	}	
+}
```
Comment (pavelnikolov) [inline internal/common/lexer_test.go:135]: Add empty line at the end, please

File: internal/common/lexer_test.go [hunk 3]
```diff
@@ -93,3 +93,43 @@ func TestConsume(t *testing.T) {
...
+	},
+}
+
+func TestMultilineString(t *testing.T) {
+	for _, test := range multilineStringTests {
+		t.Run(test.description, func(t *testing.T) {
+			lex := common.NewLexer(test.definition, test.useStringDescriptions)
+
+			err := lex.CatchSyntaxError(func() { lex.ConsumeWhitespace() })
+
```
Comment (pavelnikolov) [inline internal/common/lexer_test.go:121]: (nitpick) I'd remove this empty line

**Review by pavelnikolov — CHANGES_REQUESTED** [review]
Thank you. Can you, please fix the empty lines and I'd be happy to merge this PR.

File: internal/common/lexer.go (hunk 1 above)
Comment (suntoucha) [reply internal/common/lexer.go]: Done

File: internal/common/lexer_test.go (hunk 2 above)
Comment (suntoucha) [reply internal/common/lexer_test.go]: Done

File: internal/common/lexer_test.go (hunk 3 above)
Comment (suntoucha) [reply internal/common/lexer_test.go]: Done

**Pull request merged**